llvm/test/CodeGen/RISCV/rv64zba.ll

```diff
@@ -983,6 +983,1299 @@ define i64 @mul81(i64 %a) {
   ret i64 %c
 }
 
+
+define i64 @mul153(i64 %a) {
+; RV64I-LABEL: mul153:
```

Comment:
llvm-mca doesn't think this is an improvement for sifive-x280 https://godbolt.org/z/6ahP11xrq

Reply:
Thanks for your feedback,
tests show that 3 shXadds is _a little_ slower than mul, but for some reason GCC uses 3 shXadds - https://godbolt.org/z/oez5Pddde
And even 4 - https://godbolt.org/z/czK3Kd7bs
I'll try to find out why they do this and write later.